tests/dragonfly/cluster_test.py

```diff
@@ -396,3 +396,176 @@ async def test_cluster_replica_sets_non_owned_keys(df_local_factory):
+                "port": {replicas[1].port}
+              }}
+          ]
+        }},
+        {{
+          "slot_ranges": [
+            {{
+              "start": 10001,
+              "end": 16383
+            }}
+          ],
+          "master": {{
+            "id": "{master_ids[2]}",
+            "ip": "localhost",
+            "port": {masters[2].port}
+          }},
+          "replicas": [
+              {{
+                "id": "{replica_ids[2]}",
+                "ip": "localhost",
+                "port": {replicas[2].port}
+              }}
+          ]
+        }}
+      ]
+    """
+    await push_config(config, c_masters_admin + c_replicas_admin)
+
+    client = aioredis.RedisCluster(decode_responses=True, host="localhost", port=masters[0].port)
+
+    for i in range(10_000):
+        key = 'key' + str(i)
+        assert await client.set(key, 'value') == True
+        assert await client.get(key) == 'value'
+
+    # Make sure that getting a value from a replica works as well.
+    replica_response = await client.execute_command(
+            'get', 'key0', target_nodes=aioredis.RedisCluster.REPLICAS)
+    assert 'value' in replica_response.values()
+
+    # Push new config
+    config = f"""
+      [
+        {{
+          "slot_ranges": [
+            {{
+              "start": 0,
+              "end": 4000
+            }}
+          ],
+          "master": {{
+            "id": "{master_ids[0]}",
+            "ip": "localhost",
+            "port": {masters[0].port}
+          }},
+          "replicas": [
+              {{
+                "id": "{replica_ids[0]}",
+                "ip": "localhost",
+                "port": {replicas[0].port}
+              }}
+          ]
+        }},
+        {{
+          "slot_ranges": [
+            {{
+              "start": 4001,
+              "end": 14000
+            }}
+          ],
+          "master": {{
+            "id": "{master_ids[1]}",
+            "ip": "localhost",
+            "port": {masters[1].port}
+          }},
+          "replicas": [
+              {{
+                "id": "{replica_ids[1]}",
+                "ip": "localhost",
+                "port": {replicas[1].port}
+              }}
+          ]
+        }},
+        {{
+          "slot_ranges": [
+            {{
+              "start": 14001,
+              "end": 16383
+            }}
+          ],
+          "master": {{
+            "id": "{master_ids[2]}",
+            "ip": "localhost",
+            "port": {masters[2].port}
+          }},
+          "replicas": [
+              {{
+                "id": "{replica_ids[2]}",
+                "ip": "localhost",
+                "port": {replicas[2].port}
+              }}
+          ]
+        }}
+      ]
+    """
+    await push_config(config, c_masters_admin + c_replicas_admin)
+
+    for i in range(10_000):
+        key = 'key' + str(i)
+        assert await client.set(key, 'value') == True
+        assert await client.get(key) == 'value'
```

Comment:
About the recent timeout in CI that we had. In the future:
- such parts can be split in multiple tasks that run concurrently, all on its range or modulo
- possibly ranges that surely belong to a single node can be handled in pipelines?

When more tests are added, it can add up to hundreds of thousands or requests (its already 40k for this test). Github runners in debug mode have a very low throughput.

Reply:
Whats also under question if this load is needed. Replication tests trigger all kinds of edge cases with different values and higher load (replicate during update, etc...). Here it seems like it doesn't really matter and you verify only correctness. So you could for example check 10% random values of the range: if you have a bug it's likely to be found in one or a few passes, otherwise why check the whole range

Reply:
I did not realize we're so constraint!
Thanks for the comment, I tuned down the test.